src/Bonus.java

```diff
@@ -1,66 +1,102 @@
+    System.out.println(stringnumber);
+    // Now the fun can begin, we can try to gues the number
+    feedback(stringnumber);
 
-    }
+  }
 
-    public static void/*moet dit returntype "void" zijn of wat anders?*/ randomnumbergenerator(/*Heeft deze methode nog parameter(s) nodig?*/){
+  public static HashSet<Integer> randomnumbergenerator() {
          /*
         Vul hier de body van de methode in.
-
         Stappenplan:
         - Maak een nieuwe variabele van type Random. (Tip: Zoek op internet hoe je Random kunt gebruiken)
         - Maak een nieuwe variabele van type HashSet.
         - Schrijf een while-loop om 4 random nummers aan de hashset toe te voegen
         - return de hashset
          */
+
+    Random rand = new Random();
+    HashSet<Integer> numbers = new HashSet<>();
+    while (numbers.size() < 4) {
+      numbers.add(rand.nextInt(10)); // the random number I have set to a limit between 0 and 10
+      // this will make sence since the result will only be 4 integers long and every integer needs to be different
     }
+    return numbers;
+  }
 
-    public static void/*moet dit returntype "void" zijn of wat anders?*/ setToStringConverter(/*Heeft deze methode nog parameter(s) nodig?*/){
+  public static String setToStringConverter(HashSet<Integer> numbers) {
         /*
         Vul hier de body van de methode in.
-
         Stappenplan:
         - Maak als eerst een String variabele met als waarde een lege String. (of gebruik een StringBuilder)
         - Schrijf vervolgens een for-loop om de items in de hashset een voor een aan de String variabele toe te voegen.
         - Return de (gevulde) String variabele
          */
-    }
 
+    // This will generate a string builder, and keep hold of the value an later eventually convert it to a string
+    StringBuilder sb = new StringBuilder();
+    for (Integer number : numbers) {
+      sb.append(number);
+    }
+    return sb.toString();
+  }
 
+  public static void feedback(String stringnumber) {
+    Scanner scanner = new Scanner(System.in);
+    StringBuilder feedback = new StringBuilder();
+    System.out.println("+ = juiste nummer op de juiste plek, O = juiste nummer verkeerde plek, X = verkeerde nummer");
+    System.out.println("Doe een gok, Let op vul 4 getallen in.");
 
-    public static void/*moet dit "void" zijn of wat anders?*/ feedback(String stringnumber) {
-        Scanner scanner = new Scanner(System.in);
-        StringBuilder feedback = new StringBuilder();
-        System.out.println("+ = juiste nummer op de juiste plek, O = juiste nummer verkeerde plek, X = verkeerde nummer");
-        System.out.println("Doe een gok, Let op vul 4 getallen in.");
-        String guess = scanner.nextLine();
-        if (Objects.equals(guess, stringnumber)) {
-            System.out.println("gefeliciteerd je hebt het goed");
+    // I added a while loop that will only break out if are right formatted value has been entered
```

Comment:
heb je de lus helemaal veranderd?

Reply:
Ik heb hier een while lus en wat extra if statements voor gebruikt ook om te checken of er een valide nummer wordt gebruikt en geen letters worden gebruikt.